frontend/src/metabase/entities/model-indexes/actions.ts

```diff
@@ -1,34 +1,45 @@
 import { dissocIn } from "icepick";
 import _ from "underscore";
 
+import {
+  createModelIndex,
+  deleteModelIndex,
+  listModelIndexes,
+} from "metabase/api";
 import type Question from "metabase-lib/v1/Question";
 import type { FieldReference, Field } from "metabase-types/api";
 import type { ModelIndex } from "metabase-types/api/modelIndexes";
 import type { Dispatch } from "metabase-types/store";
 
-import { ModelIndexes } from "./model-indexes";
+import { getPkRef } from "./utils";
 
 export type FieldWithMaybeIndex = Field & {
   should_index?: boolean;
-  field_ref?: FieldReference;
+};
+
+type FieldWithIndexAndRef = Field & {
+  should_index: boolean;
+  field_ref: FieldReference;
+};
+
+const filterForRefAndShouldIndex = (
+  field: FieldWithMaybeIndex,
+): field is FieldWithIndexAndRef => {
+  return field.should_index !== undefined && !!field.field_ref;
 };
 
 export const updateModelIndexes =
   (model: Question) => async (dispatch: Dispatch, getState: any) => {
     const fields = model.getResultMetadata();
 
-    const fieldsWithIndexFlags = fields.filter(
-      (field: FieldWithMaybeIndex) => field.should_index !== undefined,
-    );
+    const fieldsWithIndexFlags = fields.filter(filterForRefAndShouldIndex);
 
     if (fieldsWithIndexFlags.length === 0) {
       return;
     }
 
     const existingIndexes: ModelIndex[] =
-      ModelIndexes.selectors.getList(getState(), {
-        entityQuery: { model_id: model.id() },
-      }) ?? [];
+      listModelIndexes.select({ model_id: model.id() })(getState()).data ?? [];
```

Comment:
These selectors are a bit of a footgun. They select the current value of the cache. However, the cache is cleared when there are no active subscribers to that endpoint + input. It's safer to fetch the data again through RTK, which will select from the cache for you if there are subscribers, but also refetch for you if there is not.

Reply:
I updated this line to dispatch an `initiate` instead, though I'm not really sure what should happen on a failure :/. From what I can see on the network tab, it's still grabbing from the store though